src/cmd/build.rs

```diff
@@ -72,27 +79,69 @@ pub fn collect_crate_metadata(working_dir: Option<&PathBuf>) -> Result<CrateMeta
     dest_wasm.push(package_name.clone());
     dest_wasm.set_extension("wasm");
 
-    Ok(CrateMetadata {
+    let crate_metadata = CrateMetadata {
+        manifest_path: manifest_path.clone(),
+        cargo_meta: metadata,
+        root_package: root_package.clone(),
         package_name,
         original_wasm,
         dest_wasm,
-    })
+    };
+    Ok(crate_metadata)
 }
 
-/// Invokes `cargo build` in the specified directory, defaults to the current directory.
+/// Builds the project in the specified directory, defaults to the current directory.
 ///
-/// Currently it assumes that user wants to use `+nightly`.
-fn build_cargo_project(working_dir: Option<&PathBuf>) -> Result<()> {
-    super::exec_cargo(
-        "build",
-        &[
+/// Uses [`cargo-xbuild`](https://github.com/rust-osdev/cargo-xbuild) for maximum optimization of
+/// the resulting Wasm binary.
+fn build_cargo_project(crate_metadata: &CrateMetadata, verbosity: Option<Verbosity>) -> Result<()> {
+    util::assert_channel()?;
+
+    // set RUSTFLAGS, read from environment var by cargo-xbuild
+    std::env::set_var(
+        "RUSTFLAGS",
+        "-C link-arg=-z -C link-arg=stack-size=65536 -C link-arg=--import-memory",
+    );
+
+    let verbosity = verbosity.map(|v| match v {
+        Verbosity::Verbose => xargo_lib::Verbosity::Verbose,
+        Verbosity::Quiet => xargo_lib::Verbosity::Quiet,
+    });
+
+    let xbuild = |manifest_path: &ManifestPath| {
```

Comment:
So `xbuild` is always used and not only when it is installed as it was before. But I think that this is actually the better decision since it is now used as library. So, good! :)